From: "Mike Rapoport (IBM)" <rppt@kernel.org>
Subject: mm: move pgtable_init() to mm/mm_init.c and make it static
Date: Mon, 20 Mar 2023 00:00:02 +0200

pgtable_init() is only called from mm_core_init().

Move it close to the caller and make it static.

Link: https://lkml.kernel.org/r/20230319220008.2138576-10-rppt@kernel.org
Signed-off-by: Mike Rapoport (IBM) <rppt@kernel.org>
Cc: David Hildenbrand <david@redhat.com>
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Cc: Mel Gorman <mgorman@suse.de>
Cc: Michal Hocko <mhocko@kernel.org>
Cc: Thomas Bogendoerfer <tsbogend@alpha.franken.de>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/include/linux/mm.h~mm-move-pgtable_init-to-mm-mm_initc-and-make-it-static
+++ a/include/linux/mm.h
@@ -2782,12 +2782,6 @@ static inline bool ptlock_init(struct pa
 static inline void ptlock_free(struct page *page) {}
 #endif /* USE_SPLIT_PTE_PTLOCKS */
 
-static inline void pgtable_init(void)
-{
-	ptlock_cache_init();
-	pgtable_cache_init();
-}
-
 static inline bool pgtable_pte_page_ctor(struct page *page)
 {
 	if (!ptlock_init(page))
--- a/mm/mm_init.c~mm-move-pgtable_init-to-mm-mm_initc-and-make-it-static
+++ a/mm/mm_init.c
@@ -2511,6 +2511,12 @@ void __init memblock_free_pages(struct p
 	__free_pages_core(page, order);
 }
 
+static void __init pgtable_init(void)
+{
+	ptlock_cache_init();
+	pgtable_cache_init();
+}
+
 /* Report memory auto-initialization states for this boot. */
 static void __init report_meminit(void)
 {
_
